### PR TITLE
Introducing type Meta to the render items of the scene

### PR DESCRIPTION
**Pull request by samcake**
The type Meta for a render item is used to indicate that a render item supervise sub items and act as a group  of render items (typically shapes).
And it is used as a 1 to 1 proxy for the entity, avatar, overlay and any other game-play object into the scene as a regular render item.
Right now, the meta specific interface of the render::Item introduce a call to collect the sub items associated to the meta.

There are more features coming from this foundation work such as:
- having an update call on metas early in the pipeline (so they could adjust properties to the rendering context)
- a more efficient way to perform pendingChanges to the meta and its sub items
- Selections and highlights

We also introduced a simple Job drawing the bounding box of the metas visible in the scene. THis is also available for debug as a checkbox in the debugDeferredLighting.js

## Test Plan
- verify that the performance are not changing compared to master
- Running the script debugDeferredLighting.js (from this PR) you should be able to see a new checkbox at the bottom allowing to display/hide the bounding boxes of the Entities and Avatars as seen by the render engine.

## To do
add voxel & light entities

### CODE REVIEW

**Comment by nbq** [comment]
Will test this soon, waiting for build

**Comment by hifi-gustavo** [comment]
Build console output for commit: 5ed3b48f9d6333a42d8a9c13c90668bef4b94dc1 available at: https://gist.github.com/a539daa7459ba4c5860665ebdb74ed7e<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9410-5ed3b48.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9410-5ed3b48.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 246dcbd0c0b343531cd3b489abdd7756af91a7fc available at: https://gist.github.com/a539daa7459ba4c5860665ebdb74ed7e<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9410-246dcbd.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9410-246dcbd.dmg

**Comment by nbq** [comment]
@samcake  Still failing one check it says so waiting to test until the error is gone

**Comment by samcake** [comment]
test this please

**Comment by hifi-gustavo** [comment]
Build console output for commit: 246dcbd0c0b343531cd3b489abdd7756af91a7fc available at: https://gist.github.com/a539daa7459ba4c5860665ebdb74ed7e<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9410-246dcbd.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9410-246dcbd.dmg

**Comment by nbq** [comment]
Tester Coal
CPU: i7 6700k
GPU: GF 970

* Tried performance with Vive and Desktop Mode and seems to work well, did smoke test, all passed
* Ran script included in this PR named `debugDeferredLighting.js` and toggled checkbox at bottom to test, clicked `Draw Meta Bounds` and it functioned properly as I saw the bounding boxs.

PASS

File: libraries/render/src/render/CullTask.h [hunk 1]
```diff
@@ -170,7 +170,7 @@ namespace render {
                 for (size_t i = 0; i < NUM_FILTERS; i++) {
                     if (_filters[i].test(itemKey)) {
                         outItems[i].template edit<ItemBounds>().emplace_back(itemBound);
-                        break;
+                      //  break;
```
Comment (Atlante45) [inline libraries/render/src/render/CullTask.h]: Is that meant to stay?

Comment (samcake) [inline libraries/render/src/render/CullTask.h]: oh yes!
good catch

**Comment by hifi-gustavo** [comment]
Build console output for commit: e3da6aa9f1812da18121074929a65da0e38185aa available at: https://gist.github.com/a539daa7459ba4c5860665ebdb74ed7e<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9410-e3da6aa.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9410-e3da6aa.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 027bd30cbc9e80495d9c691efab388c92ff39cbb available at: https://gist.github.com/a539daa7459ba4c5860665ebdb74ed7e<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9410-027bd30.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9410-027bd30.dmg

**Comment by hifi-gustavo** [comment]
Build console output for commit: 05304c72e0bae20b9751775fc474b01c2e37be98 available at: https://gist.github.com/a539daa7459ba4c5860665ebdb74ed7e<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9410-05304c7.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9410-05304c7.dmg

**Comment by Atlante45** [comment]
test this please

**Comment by hifi-gustavo** [comment]
Build console output for commit: 05304c72e0bae20b9751775fc474b01c2e37be98 available at: https://gist.github.com/a539daa7459ba4c5860665ebdb74ed7e<br/><br/>Installers available for download at: <br/>Windows: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9410-05304c7.exe<br/>OS X: http://cachefly.highfidelity.com/HighFidelity-Beta-PR9410-05304c7.dmg

**Pull request merged**